environments/react/accessibility.js

```diff
@@ -0,0 +1,218 @@
+    // key handlers). Non-interactive elements include <main>, <area>, <h1> (,<h2>, etc), <img>,
+    // <li>, <ul> and <ol>. Non-interactive WAI-ARIA roles include article, banner, complementary,
+    // img, listitem, main, region and tooltip.
+    'jsx-a11y/no-noninteractive-element-interactions': ['warn', {
+      handlers: ['onClick', 'onMouseDown', 'onMouseUp', 'onKeyPress', 'onKeyDown', 'onKeyUp'],
+    }],
+
+    // Non-interactive HTML elements indicate content and containers in the user interface.
+    // Non-interactive elements include <main>, <area>, <h1> (,<h2>, etc), <img>, <li>, <ul> and
+    // <ol>.  Interactive HTML elements indicate controls in the user interface. Interactive
+    // elements include <a href>, <button>, <input>, <select>, <textarea>.  WAI-ARIA roles should
+    // not be used to convert a non-interactive element to an interactive element. Interactive ARIA
+    // roles include button, link, checkbox, menuitem, menuitemcheckbox, menuitemradio, option,
+    // radio, searchbox, switch and textbox.
+    'jsx-a11y/no-noninteractive-element-to-interactive-role': ['warn', {
+      ul: ['listbox', 'menu', 'menubar', 'radiogroup', 'tablist', 'tree', 'treegrid'],
+      ol: ['listbox', 'menu', 'menubar', 'radiogroup', 'tablist', 'tree', 'treegrid'],
+      li: ['menuitem', 'option', 'row', 'tab', 'treeitem'],
+      table: ['grid'],
+      td: ['gridcell'],
+    }],
+
+    // Tab key navigation should be limited to elements on the page that can be interacted with.
+    // Thus it is not necessary to add a tabindex to items in an unordered list, for example, to
+    // make them navigable through assistive technology. These applications already afford page
+    // traversal mechanisms based on the HTML of the page. Generally, we should try to reduce the
+    // size of the page's tab ring rather than increasing it.
+    'jsx-a11y/no-noninteractive-tabindex': ['warn', {
+      tags: [],
+      roles: ['tabpanel'],
+    }],
+
+    // Enforce usage of onBlur over/in parallel with onChange on select menu elements for
+    // accessibility. onBlur should be used instead of onChange, unless absolutely necessary and it
+    // causes no negative consequences for keyboard only or screen reader users. onBlur is a more
+    // declarative action by the user: for instance in a dropdown, using the arrow keys to toggle
+    // between options will trigger the onChange event in some browsers. Regardless, when a change
+    // of context results from an onBlur event or an onChange event, the user should be notified of
+    // the change unless it occurs below the currently focused element.
+    'jsx-a11y/no-onchange': 'warn',
+
+    // Some HTML elements have native semantics that are implemented by the browser. This includes
+    // default/implicit ARIA roles. Setting an ARIA role that matches its default/implicit role is
+    // redundant since it is already set by the browser.
+    'jsx-a11y/no-redundant-roles': 'warn',
```

Comment:
If it's not absolutely terrible practice I would prefer keeping this at warn level. Best practices should not cause a failure in your test/deployment pipeline.